smpplib/command.py

```diff
@@ -344,7 +346,9 @@ def field_exists(self, field):
     def field_is_optional(self, field):
         """Return True if field is optional, False otherwise"""
 
-        if field in consts.OPTIONAL_PARAMS:
+        if hasattr(self, 'mandatory_fields') and field in self.mandatory_fields:
```

Comment:
Could you elaborate a bit, why `mandatory_fields` is needed? The function should return `False` unless a field is present in `OPTIONAL_PARAMS`, so it should've worked "as is". Do I maybe miss something?

Reply:
`message_state` is in OPTIONAL_PARAMS that's the issue :)

Reply:
Oh, I see. `OPTIONAL_PARAMS` is a global one, bad design... let it be a technical debt for now

Reply:
Yeah. I guess I could have reworked it to be specific to each command. But this may actually be the only case where an "optional" field becomes mandatory according to my not so extensive knowledge of the SMPP protocol specs.